mira/metamodel/template_model.py

```diff
@@ -31,11 +33,23 @@ class Config:
 
     @classmethod
     def from_json(cls, data: Dict[str, Any]) -> "Initial":
-        value = data.pop('value')
+        expression = data.pop('expression')
         concept_json = data.pop('concept')
         # Get Concept
         concept = Concept.from_json(concept_json)
-        return cls(concept=concept, value=value)
+        # If expression string isn't a float
+        try:
```

Comment:
This doesn't look correct. We need to turn the expression string into a sympy.Expr object. This requires parsing the string. In don't think instantiating SympyExprStr with a string (i.e., SympyExprStr(expression)) is a good idea.

Reply:
try:
            if expression.isdigit():
                return cls(concept=concept, expression=SympyExprStr(sympy.Integer(expression)))
            else:
                return cls(concept=concept, expression=SympyExprStr(sympy.Float(expression)))
        except ValueError:
            if len(expression) == 1:
                return cls(concept=concept, expression=SympyExprStr(sympy.Symbol(expression)))
            else:
                return cls(concept=concept, expression=SympyExprStr(sympy.sympify(expression)))

Would this work better?

Reply:
We don't need to differentiate all these cases. `sympify` works on all of them directly, i.e., 

```
In [12]: for val in [1, 2.3, '3', '4.5', 'a * b']:
    ...:     print(isinstance(sympy.sympify(val), sympy.Expr))
    ...: 
True
True
True
True
True
```

The more tricky issue is that for some expressions to be parsed correctly, we need to input a list of "local" symbols. For instance, `sympy.sympify('alpha * beta')` throws an error but `sympy.sympify('alpha * beta', locals={'beta': sympy.Symbol('beta')})` is parsed correctly (note that this would still be a problem in your code above).

So we should think about how we can ensure that deserialization is done with those symbols being accessible. I will look into this.